File: back-end/controller/racks.js
import RackPostModal from '../models/rackPost.js'

export const getRacks = async (req, res) => {
    try {
        const result = await RackPostModal.find()
        res.status(200).json(result)
    } catch (err) {
        res.status(404).json(err)
    }
}
export const getRack = async (req, res) => {
    try {
        const result = await RackPostModal.findById(req.params.id)
        res.status(200).json(result)
    } catch (err) {
        res.status(404).json(err)
    }
}
export const postRack = async (req, res) => {
    const newRack = new RackPostModal(req.body.rack)
    try {
        const result = await newRack.save()
        res.status(201).json(result)
    } catch (error) {
        res.status(404).json({ message: error.message })
    }
}
export const updateRack = async (req, res) => {
    try {
        const result = await RackPostModal.findByIdAndUpdate(req.params.id, { $set: req.body.rack }, { new: true })
        res.status(200).json(result)
    } catch (err) {
        res.status(500).json(err)
    }
}
export const deleteRack = async (req, res) => {
    try {
        const result = await RackPostModal.findByIdAndDelete(req.params.id)
        res.status(200).json(result)
    } catch (err) {
        res.status(500).json(err)
    }
}

File: back-end-python/app/routes/racks.py
from flask import request, jsonify, Blueprint
from app.models import Rack
from app.database import collection
from bson.objectid import ObjectId

racks_bp = Blueprint('racks', __name__)

@racks_bp.route('/racks', methods=['GET'])
def get_racks():
    try:
        data = list(collection.find({}))
        for doc in data:
            doc['_id'] = str(doc['_id'])
            for device in doc["devices"]:
                device['_id'] = str(device['_id'])
        return jsonify({"racks": data}), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@racks_bp.route('/racks', methods=['POST'])
def create_rack():
    try:
        data = request.json
        inserted_id = collection.insert_one(data['rack']).inserted_id
        response = {
            'message': 'Data added successfully',
            'inserted_id': str(inserted_id)
        }
        return jsonify(response), 201
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@racks_bp.route('/racks/<string:rack_id>', methods=['PATCH'])
def update_rack(rack_id):
    try:
        data = request.json
        rack_data = data['rack']
        rack_id = ObjectId(rack_id)
        collection.update_one({'_id': rack_id}, {'$set': rack_data})
        response = {
            'message': 'Data updated successfully'
        }
        return jsonify(response), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@racks_bp.route('/racks/<string:rack_id>', methods=['DELETE'])
def delete_rack(rack_id):
    try:
        rack_id = ObjectId(rack_id)
        result = collection.delete_one({'_id': rack_id})
        if result.deleted_count == 1:
            response = {
                'message': 'Data deleted successfully'
            }
            return jsonify(response), 200
        else:
            return jsonify({'message': 'Rack not found'}), 404
    except Exception as e:
        return jsonify({'error': str(e)}), 500
